### PR TITLE
Feature/app state managment

### PR DESCRIPTION
**Pull request by ostatni5**
To use in compnent:
`const store = useContext(StoreContext);`

### CODE REVIEW

File: src/lecturer/services/StoreService.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import { createContext, ReactNode, useState } from "react";
```
Comment (Petros9) [inline src/lecturer/services/StoreService.tsx:1]: is this file some sort of a broker/container to store quizes for other functions?

File: src/lecturer/components/pickQuizView/PickQuizView.tsx [hunk 2]
```diff
@@ -107,16 +111,27 @@ export function PickQuizView() {
...
+    const [title, setTitle] = useState("");
     const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
-        setName(event.target.value);
+        setTitle(event.target.value);
     };
-    const [question, setQuestion] = useState<Question[]>(questionListMock);
+    const [questions, setQuestions] = useState<Question[]>([]);
+
+
+    useEffect(() => {
```
Comment (Petros9) [inline src/lecturer/components/pickQuizView/PickQuizView.tsx:125]: why there are two functions with exactly the same name in one file?

**Review by Petros9 — APPROVED** [review]

File: src/lecturer/components/createQuestionView/CreateQuestionView.tsx [hunk 3]
```diff
@@ -206,9 +207,11 @@ export function CreateQuestionView() {
                 text: question,
                 options: options,
             };
+
+            store.questions = [...store.questions, obj];
```
Comment (Petros9) [inline src/lecturer/components/createQuestionView/CreateQuestionView.tsx:211]: so if I want to load questions to the store all I need to do is to write `store questions = [array of questions]`, right?

**Review by Petros9 — APPROVED** [review]

**Review by Derstilon — APPROVED** [review]
Should be working

File: src/lecturer/components/createQuestionView/CreateQuestionView.tsx (hunk 3 above)
Comment (ostatni5) [reply src/lecturer/components/createQuestionView/CreateQuestionView.tsx:211]: store.questions = questions

File: src/lecturer/components/pickQuizView/PickQuizView.tsx (hunk 2 above)
Comment (ostatni5) [reply src/lecturer/components/pickQuizView/PickQuizView.tsx:125]: useEffect  is react hook 
https://pl.reactjs.org/docs/hooks-effect.html

File: src/lecturer/services/StoreService.tsx (hunk 1 above)
Comment (ostatni5) [reply src/lecturer/services/StoreService.tsx:1]: The store is using react context to provide objects to all components inside him. 
https://pl.reactjs.org/docs/context.html

**Pull request merged**